findspam.py

```diff
@@ -794,6 +794,34 @@ def misleading_link(s, site):
         return False, ''
 
 
+# noinspection PyUnusedLocal,PyMissingTypeHints,PyTypeChecker
+@create_rule("text repeated in {}", title=False, body_summary=True, max_rep=10000, max_score=10000)
+def body_text_repeated(s, site):
+    """
+    Do some hacks to reduce the need for regex backtracking for this rule
+    """
+    s = s.rstrip("\n")
+    if s.startswith("<p>") and s.endswith("</p>"):
+        s = s[3:-4]
+    initial_words = regex.match(r"\A([^\W_]+)[\W_]+([^\W_]+)[\W_]+([^\W_]+)", s)
+    if not initial_words:
+        return False, ""
+    escaped_initial_words = [regex.escape(x) for x in initial_words.groups()]
+    period = regex.match(
+        r"\A%s[\W_]+%s[\W_]+%s[\W_]+(.{1,40}?)%s[\W_]+%s[\W_]+%s(?=$|[\W_])" % (
+            tuple(escaped_initial_words * 2)), s)
+    if not period:
+        return False, ""
+    period_words = regex.split(r"[\W_]+", period.groups(0)[0])
+    escaped_words = escaped_initial_words + [
+        regex.escape(x) for x in period_words]
+    repeats_regex = r"\A(" + r"[\W_]+".join(escaped_words) + r"[\W_]*){10,}"
+    repeats = regex.match(repeats_regex, s)
```

Comment:
Same request here regarding `regex.cache_all(False)` prior to this `regex.match()` and `regex.cache_all(True)` after it.

Reply:
Obviously, it would be possible to use only a single pair of `regex.cache_all(False)` and `regex.cache_all(True)` to cover both post-specific uses of a regular expression.

This also touches on the larger issue of how we use the `regex` package's cache, particularly within the detections. The larger issue is that we don't want to be using it for regexes which are individually generated specific to each post, but we do want any static regular expressions not to have to be recompiled for every use (e.g. through using the `regex` package's cache). There are a couple of different approaches which we could take in general. I'll open an Issue to cover/discuss the topic.